pkg/scheduler/schedule_one.go

```diff
@@ -91,11 +91,35 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 
 	schedulingCycleCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	scheduleResult, assumedPodInfo := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, podsToActivate, start)
-	if scheduleResult.FeasibleNodes == 0 {
+
+	scheduleResult, assumedPodInfo, err := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, start)
+	if err != nil {
+		switch scheduleResult.reason {
+		case v1.PodReasonUnschedulable:
+			metrics.PodUnschedulable(fwk.ProfileName(), metrics.SinceInSeconds(start))
+		case v1.PodReasonSchedulerError:
+			metrics.PodScheduleError(fwk.ProfileName(), metrics.SinceInSeconds(start))
+		}
+
+		if scheduleResult.unreserve {
+			// trigger un-reserve to clean up state associated with the reserved Pod
+			fwk.RunReservePluginsUnreserve(ctx, state, assumedPodInfo.Pod, scheduleResult.SuggestedHost)
+			if forgetErr := sched.Cache.ForgetPod(assumedPodInfo.Pod); forgetErr != nil {
+				klog.ErrorS(forgetErr, "Scheduler cache ForgetPod failed")
+			}
+		}
+
+		sched.FailureHandler(ctx, fwk, assumedPodInfo, err, scheduleResult.reason, scheduleResult.nominatingInfo)
 		return
 	}
 
+	// At the end of a successful scheduling cycle, pop and move up Pods if needed.
```

Comment:
Moved this out of schedulingCycle, podsToActivate is a value of pointer, I think it's ok.